django/contrib/admin/views/main.py

```diff
@@ -308,7 +308,9 @@ def get_ordering(self, request, queryset):
                     order_field = self.get_ordering_field(field_name)
                     if not order_field:
                         continue  # No 'admin_order_field', skip it
-                    if hasattr(order_field, 'as_sql'):
+                    if isinstance(order_field, OrderBy):
+                        ordering.append(order_field)
```

Comment:
`OrderBy` should also support reverse ordering.

Reply:
We already have a  `reverse_ordering()`. Moreover this solution mutates `OrderBy` expression, so maybe:
```python
if isinstance(order_field, OrderBy):
    if pfx == '-':
        order_field = order_field.copy()
        order_field.reverse_ordering()
    ordering.append(order_field)
```

Reply:
Thanks for the description. I replaced your suggested code block.